Firestore/core/test/unit/api/aggregate_query_test.cc

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "Firestore/core/src/api/query_core.h"
+#include "Firestore/core/src/util/status.h"
+#include "gtest/gtest.h"
+
+#define private public
+#include "Firestore/core/src/api/aggregate_query.h"
+
+namespace firebase {
+namespace firestore {
+
+using model::AggregateAlias;
+using util::Status;
+
+namespace api {
+
+class MockAggregateQuery : public AggregateQuery {
```

Comment:
We should have a broader team discussion on this. I think we are holding back our ability to unit test without virtual methods.

Reply:
Fair enough. Can you just add a comment then at the declaration site explain why the method is marked "virtual"?

Reply:
After looking at the code in more detail I am more sympathetic to your approach to making the one method `virtual`. However, I'd recommend to use gmock (which is built into googletest) instead of rolling your own hand-built mock object. See https://github.com/google/googletest/tree/main/googlemock for details.

Reply:
Comment added to the declaration site about use and future removal of `virtual`.

Thanks for pointing out gmock. It's worth using and learning, even if it make this code more complex.